File: src/components/about/Interfaces.ts

export interface FormationCategory{
  categoryTitle: string;
  formations: Formation[];
}

export interface Formation{
  name: string;
  description: string;
  year: string;
  place: string; 
}

export interface TechnosCategory{
  categoryTitle: string;
  technos: Techno[];
}

export interface Techno {
  title: string;
  names: string[];
}

export interface LanguageCategory {
  categoryTitle: string;
  languages: Language[];
}

export interface Language {
  name: string;
  level: string;
}

export interface ExperienceCategory{
  categoryTitle:string;
  experiences: Experience[];
}

export interface Experience{
  title: string;
  description: string;
  technos: string[];
}

File: src/components/about/interfaces.rs
// use serde::{Deserialize, Serialize};






pub struct LanguageCategory {
  pub categoryTitle: String,
  pub languages: Vec<Language>,
}

pub struct Language {
  pub name: String,
  pub level: String,
}

pub struct ExperienceCategory{
  pub categoryTitle:String,
  pub experiences: Vec<Experience>,
}

pub struct Experience{
  pub title: String,
  pub description: String,
  pub technos: Vec<String>,
}
